src/install.ts

```diff
@@ -26,9 +26,9 @@ export type OnDemandDependencies = Map<string, string>;
  * @see https://github.com/nodejs/node/issues/31803
  */
 export const assertResolvable = async (id: string) => {
-  await new Promise(async (resolve, reject) => {
+  await new Promise<void>(async (resolve, reject) => {
     exec(
-        `${process.execPath} -e "require.resolve(process.env.ID)"`,
+        `"${process.execPath}" -e "require.resolve(process.env.ID)"`,
```

Comment:
The quotes are necessary for this to work on windows where node path might be `C:\Program Files\NodeJS\node.exe` (note the space in "Program Files" which necessitates the quotes).